analyses/cell-type-ewings/template_notebooks/singler-workflow/01-singler-results.Rmd

```diff
@@ -0,0 +1,314 @@
+The top 7 cell types are shown for each reference used and all other cells are lumped together. 
+
+```{r, fig.height=8, fig.width=8}
+plot_faceted_umap(classification_df, singler_lumped)
+```
+
+
+## Comparison between `SingleR` and `AUCell`
+
+Below we show the cells that are annotated as tumor by both methods, just `AUCell` and just `SingleR` along with all normal cells. 
+
+```{r}
+plot_faceted_umap(classification_df, consensus)
+```
+
+We then calculate the confusion matrix between tumor and normal cells called in `SingleR` and `AUCell`. 
+The rows here correspond to the `AUCell` annotation and columns are from `SingleR`. 
+
+```{r}
+caret_df <- classification_df |> 
+  dplyr::filter(singler_tumor_normal != "unknown") |> 
+  dplyr::mutate(singler_tumor_normal = droplevels(singler_tumor_normal, "unknown"))
+
+caret::confusionMatrix(
+  table(
+    caret_df$aucell_annotation,
+    caret_df$singler_tumor_normal
+  )
+)
+```
+
+
+## Validate output from `SingleR`
+
+The next three sections will evaluate the annotations output obtained from using `SingleR` by looking at expression of marker genes and gene set scores. 
+
+The gene expression for all marker genes for tumor cells, endothelial cells, mesenchymal-like cells, and immune cells was summed together for each cell. 
+Marker genes were obtained from [Visser et al.](https://doi.org/10.1158/2767-9764.CRC-23-0027). 
+
+Gene set scores are determined by calculating the mean expression of all genes in a given gene set. 
+Scores are calculated for the following gene sets from `MsigDb`:
+
+  - [`ZHANG_TARGETS_OF_EWSR1_FLI1_FUSION`](https://www.gsea-msigdb.org/gsea/msigdb/human/geneset/ZHANG_TARGETS_OF_EWSR1_FLI1_FUSION.html)
+  - [`RIGGI_EWING_SARCOMA_PROGENITOR_UP`](https://www.gsea-msigdb.org/gsea/msigdb/human/geneset/RIGGI_EWING_SARCOMA_PROGENITOR_UP.html?ex=1)
+  - [`SILIGAN_TARGETS_OF_EWS_FLI1_FUSION_DN`](https://www.gsea-msigdb.org/gsea/msigdb/cards/SILIGAN_TARGETS_OF_EWS_FLI1_FUSION_DN)
+
+For both marker gene expression and gene set scores, the following plots will be shown: 
+
+- Heatmap showing gene expression or gene set scores across all cells. 
+Here the rows correspond to the gene list and the columns are the cells. 
+The annotations are shown below the heatmap. 
+- Density plot showing gene expression or gene set scores across all cells. 
+Each row is a cell type and the expression or score is plotted on the x-axis. 
+
+
+### Tumor vs. Normal 
+
+In this section we show just the cells that are considered tumor and normal, lumping all non-tumor cell types together. 
+
+**Marker gene expression**
+
+```{r}
+full_celltype_heatmap(classification_df, marker_gene_columns, "singler_tumor_normal")
+```
+
+
+```{r}
+plot_density(classification_df,
+             "tumor_sum",
+             "singler_tumor_normal")
+```
+
+**Gene set scores**
+
+```{r}
+full_celltype_heatmap(classification_df, geneset_columns, "singler_tumor_normal")
+```
+
+```{r}
+geneset_columns |> 
+  purrr::map(\(column){
```

Comment:
Just noting that the code here is writing out the list names, which I think can be avoided by capturing the `map` output:
```suggestion
plots <- geneset_columns |> 
  purrr::map(\(column){
```

I don't _think_ that would prevent the plots from printing, but if it did you could use `patchwork:: wrap_plots()` on the resulting list.